### ISSUE TITLE
Actions: Adding token logo change feature to 'Edit Colony details' action type

### ISSUE THREAD

**Issue by melyndav** [JoinColony/colonyCDapp#1632]
## Description

Currently, there is no way for users to change a Colony's token logo once created in the onboarding flow for a Colony's creation. The scope of this PR updates the existing 'Edit Colony details' action type to enable this functionality.

## Scope

- Update the existing 'Edit Colony details' action type and add a new row called 'token logo'. 

By default, the logo shown in the avatar will be either the one uploaded by the user during the 'Create a Colony' onboarding flow or the default assigned blockie if the user did not upload a token logo during this flow. 

_It's important when testing this step when in active dev, that this is tested within the onboarding flow.

Also important: when the action goes to a motion, the correct and updated logo should be displayed. _ 

<img width="507" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/249de7a8-b01d-400f-8a32-02b5627427f4">

Figma link: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=1678-168411&mode=design&t=V9WB8knufbS8936p-4

- Interaction states

The hover state on the 'change logo' text is text-blue-400 for desktop. On select, a custom model is triggered, allowing the issue to upload a new avatar (this is the same behaviour as the 'Change Colony logo' model but with custom text within the model.

Full flow for desktop and mobile: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=18821-229063&mode=design&t=V9WB8knufbS8936p-4

_It's important that the mobile version is added and design matches Figma - particular with the model._

**Comment by rdig** [comment]
Also, @ArmandoGraterol why did you start working on this, as this wasn't planned for this cycle: https://github.com/orgs/JoinColony/projects/40 ?

And there were more priority issues left in the "Todo" stack

**Comment by rdig** [comment]
@ArmandoGraterol and @arrenv We need to figure out a way forward here, otherwise this issue will be closed.

**Comment by arrenv** [comment]
> @ArmandoGraterol and @arrenv We need to figure out a way forward here, otherwise this issue will be closed.

Agreed, and there is the intention of having a dedicated action for token management, however, adding a new action is more work and one just for the logo seems unneeded.

The intention of this is that allows for a logo that is added during the creation process to be modified. Otherwise, it exists unchangeable.

Externally added token logos are not changeable.

**Comment by rdig** [comment]
> The intention of this is that allows for a logo that is added during the creation process to be modified. Otherwise, it exists unchangeable.

So this is the point I want to discuss specifically, I'm ok with implementation. Most (all?) tokens, once created, either by our contracts, or by others, cannot have any details changed (name, symbol, decimals), so why have the avatar different ?

In my opinion, having the avatar changeable after the fact is a security risk, as it has the potential to highjack / spam the token via it's avatar image.

**Comment by arrenv** [comment]
> So this is the point I want to discuss specifically, I'm ok with implementation. Most (all?) tokens, once created, either by our contracts, or by others, cannot have any details changed (name, symbol, decimals), so why have the avatar different ?

The details are on-chain, however, the logo is only stored in our database, the same with other services, icons are stored in third party services. We also don't want to have to force the logo in the creation step as an additional block in onboarding.

> In my opinion, having the avatar changeable after the fact is a security risk, as it has the potential to highjack / spam the token via it's avatar image.

In terms of security, as you say, the name, symbol and address does not change. It will also show the logo in our UI, not in their wallet or other blockchain apps.

While not specifically discussed I think the token icon should prioritize existing token logo services, then fallback to our hosted icon, then fallback to blockie.

Just to add to the discussion, the reason why we want the token icon stored in the app is because getting accepted on trusted token logo services requires a lot more of an approval process and a lot of criteria. But we feel within our UI, having a brand logo adds to the experience and personalization of a Colony.

**Comment by rdig** [comment]
> While not specifically discussed I think the token icon should prioritize existing token logo services, then fallback to our hosted icon, then fallback to blockie.

This increases the scope of this to full blown feature, but I'll happily discuss the required specs

> Just to add to the discussion, the reason why we want the token icon stored in the app is because getting accepted on trusted token logo services requires a lot more of an approval process and a lot of criteria. But we feel within our UI, having a brand logo adds to the experience and personalization of a Colony.

I'm not arguing us not having token logos, I'm arguing that they shouldn't be easily changeable. That's why the process for the logo services that you mentioned is so arduous. It takes a degree of trust and security, hence not having this easily changeable is a requirement.

If this really a must-have feature, we'll put it in of course, but I do want to point out that this will make us liable for whatever users end up uploading in there when token spoofs will be attempted

**Comment by arrenv** [comment]
> If this really a must-have feature, we'll put it in of course, but I do want to point out that this will make us liable for whatever users end up uploading in there when token spoofs will be attempted.

We allow this anyway during the Colony creation process, users can upload whatever they want. What I think would help here though is a flag that this was a token created by the Colony, which we have discussed and do have a variation of in the design.

So, I will review that and perhaps it can also be included as a part of this scope.
